backend/src/rtp.rs

```diff
@@ -1541,6 +1554,48 @@ pub struct Endpoint {
     rtx_sender: RtxSender,
 }
 
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum EndpointError {
+    #[error("SRTP decryption error: {0}")]
+    SrtpDecryptionError(String),
+
+    #[error("SRTP encryption error: {0}")]
+    SrtpEncryptionError(String),
+
+    #[error("RTCP sender SSRC not set")]
+    RtcpSenderSsrcNotSet,
+
+    #[error("Invalid outgoing SRTCP index")]
+    InvalidOutgoingSrtcpIndex,
+
+    #[error("Incoming SSRC state not found for SSRC: {0}")]
+    IncomingSsrcStateNotFound(Ssrc),
+
+    #[error("Transport-CC receiver error: {0}")]
+    TccReceiverError(String),
+
+    #[error("Transport-CC sender error: {0}")]
+    TccSenderError(String),
+
+    #[error("Max received TCC sequence number error: {0}")]
+    MaxReceivedTccSeqnumError(String),
+
+    #[error("RTX sender error: {0}")]
+    RtxSenderError(String),
+
+    #[error("Undetermined Endpoint error: {0:?}")]
+    UndeterminedError(String),
+}
+
+//TODO: Develop some way for performing cross error conversions
+// impl<T> std::convert::From<T> for EndpointError
+// where T: std::string::ToString
+// {
+//     fn from(error: T) -> Self {
+//         Self::UndeterminedError(error.to_string())
+//     }
+// }
```

Comment:
Maybe this block isn't necessary either. I am not sure who will consume this or if there is a need.

Reply:
I put that out perhaps due to need for interconversion among Error types if in future someone need to. But this can be scraped out completely if we don't plan to bring that here.